sui-execution/latest/sui-adapter/src/temporary_store.rs

```diff
@@ -778,7 +779,11 @@ impl<'backing> TemporaryStore<'backing> {
     /// All objects will be updated with their new (current) storage rebate/cost.
     /// `SuiGasStatus` `storage_rebate` and `storage_gas_units` track the transaction
     /// overall storage rebate and cost.
-    pub(crate) fn collect_storage_and_rebate(&mut self, gas_charger: &mut GasCharger) {
+    pub(crate) fn collect_storage_and_rebate(
+        &mut self,
+        gas_charger: &mut GasCharger,
+    ) -> TransactionRebateInfo {
```

Comment:
Is it possible to introduce new functions that returns this information from the `TemporaryStore` that only get called from replay?  I know we can't avoid modifying existing execution layers altogether, but where possible, I want to make sure the modification is isolated, and we don't incur costs during normal execution to benefit local tools.